### DISCUSSION TITLE
Disable Auto-Sign up with unknown emails

### DISCUSSION THREAD

**Discussion by VaZark** [nextauthjs/next-auth#8024] [Help]
Is there a native way to disable automatic signup when the app uses `EmailProvider` ? The service I'm building is exclusively for internal use. 

So I'd like to restrict signup to exclusively to invited users (invites should be easy to implement)

**Comment by nhuethmayr** [comment]
Hi @VaZark,

this is a simplified version of what my `signIn` callback function looks like.

```
signIn: async ({ user, email }) => {
      if (email?.verificationRequest) {
        //if its a verification request (aka first sign-up)
        //it mustn't be from a blocked domain
        const domain = getDomainFromEmail(user.email)
        const blockedEmailDomain = freeEmailDomains.includes(domain)

        if (blockedEmailDomain) {
          return new URL(
            '/sign-up?error=PublicEmailBlocked',
            process.env.NEXTAUTH_URL
          ).toString()
        }
      }
    }
```

instead of using the `free-email-domains` npm package, you can simply allow only a single domain and `return false` for all others.
That way you have tight control over what accounts can be created or not.
